server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -504,6 +512,21 @@ public synchronized IndexShard createShard(
                 }
             };
             Directory directory = directoryFactory.newDirectory(this.indexSettings, path);
+            Directory remoteDirectory = null;
+            RemoteStoreRefreshListener remoteStoreRefreshListener = null;
+            if (this.indexSettings.isRemoteStoreEnabled()) {
+                try {
+                    Repository repository = repositoriesService.repository("dragon-stone");
```

Comment:
Is there any reason not to put the repository name into the index settings now? I know this isn't finalized as it is still behind an experimental feature flag, but it seems like you could replace the boolean with a repository name to avoid hardcoding a name.

Reply:
@andrross We may not go ahead with index specific repositories. A repository for remote store will be created for the cluster and data of each remote_store enabled index will be stored in the same repository (open to suggestions). We are tracking the required change here: https://github.com/opensearch-project/OpenSearch/issues/3286

Reply:
In the current change, we hard-coded repository name as `dragon-stone` but we would be making it configurable (as part of https://github.com/opensearch-project/OpenSearch/issues/3286). IMO, it is better to get this input from user instead of using a pre-defined name. This is mostly to avoid conflict where the repository with the same name already exists.

Reply:
> @sachinpkale what is the reason to couple `IndexService` with `RepositoriesService`? I think the `RemoteDirectoryFactory` should make the decision what kind of the repository should be created (by name or by index/cluster settings).

Here, we are passing RepositoriesService to fetch repository details (repository creation is not done as a part of this flow)

Reply:
> > IMO, it is better to get this input from user
> 
> Why would the repository name need to differ from the cluster name or UUID? I don't imagine we expect a many repositories to one cluster relationship? I also think this is a nice way to link the decoupled remote storage to the cluster?

It would create conflict if user already has a repository name with cluster UUID which is being used for snapshots.

Reply:
> I understand that, my question is what prevents `RemoteDirectoryFactory` to use `RepositoriesService` to create remote directory? Why `IndexService` should be doing it?

If we add logic of fetching repository name in `RemoteDirectoryFactory`, it would tightly couple remote directory with the repository name. It should be caller's responsibility to provide repository to be used. This makes it extensible to create remote directory instance for another repository as well (For example, if we want to re-use remote directory for snapshots in the future).

Reply:
> > Why would the repository name need to differ from the cluster name or UUID? I don't imagine we expect a many repositories to one cluster relationship? I also think this is a nice way to link the decoupled remote storage to the cluster?
> 
> It would create conflict if user already has a repository name with cluster UUID which is being used for snapshots.

@nknize Can we discuss this part over https://github.com/opensearch-project/OpenSearch/issues/3286 as it will be handling making the name configurable?

Reply:
> If the repository name is provided by caller

Are you suggesting to pass `RepositoriesService` as well as `RepositoryName` to newDirectory method?

Reply:
Just repository name, the factory could inject `RepositoriesService` (or any other means to reference the source)

Reply:
Tracking it here: https://github.com/opensearch-project/OpenSearch/issues/3547

Reply:
> Why would the repository name need to differ from the cluster name or UUID? I don't imagine we expect a many repositories to one cluster relationship? I also think this is a nice way to link the decoupled remote storage to the cluster?

@nknize I changed the repository name to cluster UUID.